services/QuillLMS/spec/services/classlink_integration/user_persister_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ClasslinkIntegration::UserPersister do
+  let(:incoming_user) do
+    {
+      email: 'foo@bar.com',
+      name: 'John W Booth',
+      password: 'Booth',
+      role: 'student',
+      signed_up_with_google: false,
+      username: 'johnbooth',
+      sourced_id: 'external-123'
+    }
+  end
+
+  describe '#run' do
+    let(:classlink_instance) { create(:classlink_instance) }
```

Comment:
Since the setup for persister is the same parameters, you could refactor out a `subject { described_class.new(incoming_user:, existing_provider_users:, classlink_instance_id: classlink_instance.id) }`

Then your two tests become:

`expect { subject }`

This communicates that you're testing the same structure of `subject` in each test.

Reply:
Similar to above, the structure of `persister` is subtly different between the tests

Reply:
I'm not sure I follow.  They look the same to me:

        persister = described_class.new(incoming_user:, existing_provider_users:, classlink_instance_id: classlink_instance.id)
        persister = described_class.new(incoming_user:, existing_provider_users:, classlink_instance_id: classlink_instance.id)